src/hooks/useErc20.ts

```diff
@@ -59,7 +65,92 @@ export function useErc20(params: Params): Return {
           setIsLoading(false)
         })
     }
-  }, [address, dispatch, erc20, networkId])
+  }, [address, erc20, networkId, saveErc20s])
 
   return { isLoading, error, value: erc20 }
 }
+
+export type UseMultipleErc20Params = { addresses: string[]; networkId?: Network }
+
+/**
+ * Fetches multiple erc20 token details for given network and addresses
+ * More efficient method to fetch many tokens at once, and avoid unnecessary re-renders
+ *
+ * Tries to get it from globalState.
+ * If not found, tries to get it from the network.
+ * Saves to globalState if found.
+ *`value` is an object with the `address` as key and it's value is either `null` when not found or the erc20
+ * Returns `isLoading` to indicate whether fetching the value
+ * Returns `error` with the error messages, if any.
+ */
+export function useMultipleErc20(params: UseMultipleErc20Params): Return<Record<string, SingleErc20State>> {
+  const { addresses, networkId } = params
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState('')
+
+  const getMultipleErc20 = useGetMultipleErc20sFromGlobalState(networkId)
+  const saveErc20s = useSaveErc20sToGlobalState(networkId)
```

Comment:
mm, this is not how you get the state and change the state, right?
persisting a token should be just dispatching an action. Maybe I'm missing sth

Reply:
Yes, that is. The hook `useSaveErc20sToGlobalState` returns a function that encapsulates the logic of dispatching the action.

Probably over engineered :sweat_smile: 

What do you suggest? Not have the hook and access the state directly?

Reply:
This hook is only to update the state `useSaveErc20sToGlobalState`. I renamed it to a shorted version - as well as the other hooks on this module. It's now `useSaveErc20s`.

I kept it returning a callback. The other hooks are now returning the object directly instead of the callback